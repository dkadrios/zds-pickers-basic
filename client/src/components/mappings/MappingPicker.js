import React from 'react'
import PropTypes from 'prop-types'
import Select from '@material-ui/core/Select'

const MappingPicker = (props) => {
  const { className, menuSource, onChange, ...rest } = props

  const dropdownProps = {
    onChange: ({ target }) => onChange(target.value),
    disableUnderline: true,
  }
  return (
    <Select
      {...dropdownProps}
      {...rest}
    >
      {menuSource}
    </Select>
  )
}

MappingPicker.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  menuSource: PropTypes.array.isRequired,
  className: PropTypes.string,
}
MappingPicker.defaultProps = {
  className: null,
}

export default MappingPicker
