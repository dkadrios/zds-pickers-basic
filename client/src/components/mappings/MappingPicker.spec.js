import React from 'react'
import MappingPicker from './MappingPicker'

const onChange = jest.fn()

const props = {
  value: '2',
  onChange,
  menuSource: [],
}

describe('MappingPicker tests', () => {
  beforeEach(jest.clearAllMocks)

  it('renders correctly', () => {
    shallowExpect(MappingPicker)(props).toMatchSnapshot()
  })

  it('onChange works correctly', () => {
    const value = '32'
    shallow(<MappingPicker {...props} />).simulate('change', { target: { value } })
    expect(onChange).toHaveBeenCalledWith(value)
  })
})
