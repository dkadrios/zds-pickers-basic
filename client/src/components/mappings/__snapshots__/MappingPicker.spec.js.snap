// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MappingPicker tests renders correctly 1`] = `
<WithStyles(WithFormControlContext(Select))
  disableUnderline={true}
  onChange={[Function]}
  value="2"
/>
`;
